### PR TITLE
Add _meta field to ilm policy

### PR DESCRIPTION
**Pull request by gaobinlong**
Relates to #70755.

The main changes of this PR are:
1. Add an optional `_meta` field to ILM policy.
2. Add some test code about the change.
3. Update the doc of `Create or update lifecycle policy API`.

### CODE REVIEW

**Comment by gaobinlong** [comment]
@dakrone, can you help to take a look at this PR?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicy.java [hunk 1]
```diff
@@ -79,6 +98,11 @@ public LifecyclePolicy(StreamInput in) throws IOException {
         type = in.readNamedWriteable(LifecycleType.class);
         name = in.readString();
         phases = Collections.unmodifiableMap(in.readMap(StreamInput::readString, Phase::new));
+        if (in.readBoolean()) {
+            this.metadata = in.readMap();
+        } else {
+            this.metadata = null;
+        }
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicy.java:105]: This needs some version control logic, something like:

```java
if (in.getVersion().onOrAfter(Version.V8_0_0)) {
    this.metadata = in.readMap();
} else {
    this.metadata = null;
}
```

(`readMap()` and `writeMap(...)` can return null and handle null values)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicy.java [hunk 2]
```diff
@@ -106,6 +133,12 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeNamedWriteable(type);
         out.writeString(name);
         out.writeMap(phases, StreamOutput::writeString, (o, val) -> val.writeTo(o));
+        if (this.metadata == null) {
+            out.writeBoolean(false);
+        } else {
+            out.writeBoolean(true);
+            out.writeMap(this.metadata);
+        }
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicy.java:141]: Same here about version control, something like:

```java
if (out.getVersion().onOrAfter(Version.V8_0_0)) {
    out.writeMap(this.metadata);
}
```

**Review by dakrone — CHANGES_REQUESTED** [review]
Thanks for working on this @gaobinlong! I left a couple of comments, we need to add BWC version checks for the serialization

**Comment by dakrone** [comment]
@elasticmachine ok to test

**Comment by gaobinlong** [comment]
@dakrone, thanks for your review, I have added some BWC code and all checks have passed yet.

**Comment by dakrone** [comment]
@elasticmachine test this please

**Review by dakrone — APPROVED** [review]
LGTM, thanks for adding this @gaobinlong! I'll merge this and work on backporting it.

**Pull request merged**